instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/MessagingAttributesExtractor.java

```diff
@@ -33,7 +34,15 @@ public final void onStart(AttributesBuilder attributes, Context parentContext, R
       set(attributes, SemanticAttributes.MESSAGING_TEMP_DESTINATION, true);
       set(attributes, SemanticAttributes.MESSAGING_DESTINATION, TEMP_DESTINATION_NAME);
     } else {
-      set(attributes, SemanticAttributes.MESSAGING_DESTINATION, destination(request));
+      if (request.getClass().getName().equals("org.apache.rocketmq.common.message.MessageClientExt")) {
```

Comment:
You'll have to find some other way of doing this. You should not add any rocketmq specific stuff to the common code.